### ISSUE TITLE
NumPy brain crash

### ISSUE THREAD

**Issue by joreiff** [PyCQA/astroid#695]
### Steps to reproduce
Create an AST for the following code snippet, e.g. using pylint:
```
if False:
    def linspace():
        pass
```

### Current behavior
```
Traceback (most recent call last):
  File "...\site-packages\pylint\lint.py", line 1119, in get_ast
    return MANAGER.ast_from_file(filepath, modname, source=True)
  File "...\site-packages\astroid\manager.py", line 91, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "...\site-packages\astroid\builder.py", line 136, in file_build
    return self._post_build(module, encoding)
  File "...\site-packages\astroid\builder.py", line 160, in _post_build
    module = self._manager.visit_transforms(module)
  File "...\site-packages\astroid\manager.py", line 69, in visit_transforms
    return self._transform.visit(node)
  File "...\site-packages\astroid\transforms.py", line 89, in visit
    module.body = [self._visit(child) for child in module.body]
  File "...\site-packages\astroid\transforms.py", line 89, in <listcomp>
    module.body = [self._visit(child) for child in module.body]
  File "...\site-packages\astroid\transforms.py", line 54, in _visit
    visited = self._visit_generic(value)
  File "...\site-packages\astroid\transforms.py", line 61, in _visit_generic
    return [self._visit_generic(child) for child in node]
  File "...\site-packages\astroid\transforms.py", line 61, in <listcomp>
    return [self._visit_generic(child) for child in node]
  File "...\site-packages\astroid\transforms.py", line 67, in _visit_generic
    return self._visit(node)
  File "...\site-packages\astroid\transforms.py", line 57, in _visit
    return self._transform(node)
  File "...\site-packages\astroid\transforms.py", line 39, in _transform
    if predicate is None or predicate(node):
  File "...\site-packages\astroid\brain\brain_numpy.py", line 499, in _looks_like_numpy_function
    return node.name == func_name and node.parent.name == numpy_module_name
AttributeError: 'If' object has no attribute 'name'
```

### Expected behavior
No crash.

### astroid version
2.2.5

**Comment by usagitoneko97** [comment]
This happen because of this [line](https://github.com/PyCQA/astroid/blob/2.2/astroid/brain/brain_numpy.py#L499) in brain_numpy, which I think has been fixed in 2.3.0.dev0 with better `numpy` brain inference.

Anyway, that line can be change to `node.parent.frame().name` instead of `node.parent.name` for a quick fix because it assume every function that is called `linspace` will always have parent that is of type Module.

**Comment by joreiff** [comment]
Thanks for the quick response! Using `node.parent.frame().name` does indeed prevent the crash. Looking forward to version 2.3.0.
